### ISSUE TITLE
Some questions about your implementation.

### ISSUE THREAD

**Issue by GorkaUrbizu** [cccwam/rc2020_electra#1]
Hi!

I'm trying to train an Electra Language Model on my own dataset, and I found your code very interesting, mainly because, you implemented gradient accumulation, a feature which I might need, to be able to keep the batch size of the original paper.

But I have some questions about your implementation, which I'd like to ask you directly...

1. Is this code ready to use for my own dataset and tokenizer (on another language)?
2. You re-implemented the small variant of ELECTRA. I'm interested on training an Electra-base. Other than all the different hyper-parameters, is there any other detail about implementation, that I should take into account (and you are aware of)?
3. I'm planning to use a TPU (using XLA), and it looks like the code is ready to run on TPUs, am I right?

Thanks,

Gorka

**Comment by cccwam** [comment]
Hi,

Sorry for the late answer.

1- For the tokenizer, I included the script to train your own tokenizer based on models available from HuggingFace Tokenizers
For the dataset, I used HuggingFace Datasets as usual NLP datasets are available on this library. You will have to adapt the code for this, the best approach may be to adapt this [function](https://github.com/cccwam/rc2020_electra/blob/e6b5acc8df5c6a724d7d1fa1bf550a9a90a6b6e2/src/data/make_dataset_document_electra.py#L49).

2- As far as I remember, the only differences are related to the hyperparameters. The main issue would be the GPU memory since the model is bigger.

3- I didn't use TPU but the training loop is from Huggingface Transformers, which is supposed to work also on TPU.

Let me know if you need more information
